### PR TITLE
HW4

### PR DESCRIPTION
**Pull request by Amudah41**

### CODE REVIEW

File: hw4/tests/task41_read_file_test.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+#     monkeypatch.setattr(
+#         hw4.tasks.task41_read_file,
+#         "open",
+#         mock
+#     )
+
+
+# def test_read_magic_number_all_exceptions_positive_test(my_path):
+#     assert read_magic_number_all_exceptions(my_path) == True
```
Comment (Amudah41) [inline hw4/tests/task41_read_file_test.py]: Tried to make the test with mock to reject using files, but interpreter wrote the mistake
E AttributeError: <module 'hw4.tasks.task41_read_file' from ... > has no attribute 'open'.
Is it possible to solve this problem and, if yes, how? My attempts were failed.

File: hw4/tasks/task44_doctest.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    for i in range(1, n + 1):
+        buffer.append(
+            "fizz" * int(i % 3 == 0)
+            + " " * int(i % 15 == 0)
+            + "buzz" * int(i % 5 == 0)
+            + str(i) * (i % 3 != 0 and i % 5 != 0)
+        )
+
+    return buffer
+
```
Comment (Amudah41) [inline hw4/tasks/task44_doctest.py:101]: Which approach of fizzbuzz implementation is better?

**Review by AlekseiAfanasev — APPROVED** [review]

File: hw4/tests/task41_read_file_test.py (hunk 1 above)
Comment (AlekseiAfanasev) [reply hw4/tests/task41_read_file_test.py]: try to mock __builtin__.open

Comment (Amudah41) [reply hw4/tests/task41_read_file_test.py]: Done

**Pull request merged**